### ISSUE TITLE
conda cannot be built with conda-build 3

### ISSUE THREAD

**Issue by gabm** [conda/conda-build#2318]
I can't build conda with CB3 3.0.11.That used to work in CB2...

I am not sure, but I think there are two bugs regarding the handling of the activate/deactivate scripts.

### idea behind the (special) current handling of conda
just repeating what I understood to make sure that my conlcusions are right

- `always_include_files` will include the activate/deactivate/conda files, which will be produced by the build process.
- initially after build-env creation they will be symlinks and in build.sh they will be unlinked
- build.sh produces new activate/deactivate and conda scripts
- because always_include_files is specified they are actually included into the package and not dropped

### Problem1: always_include_files doesn't work anymore for activate/deactivate/conda

When i build the recipe i get errors for activate/deactivate/conda:

```
ERROR:conda_build.utils:invalid recipe path: /root/miniconda3/conda-bld/conda_1503661965202/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin/conda
```

It seems the new logic in https://github.com/conda/conda-build/blob/38dc7778f17035956c3f1d8f311978152282ce59/conda_build/utils.py#L770 checks whether these are available, but at the time of checking they are not yet. The problem is, that as a result of not being present, the 3 files get dropped from the `always_include_files`-list.

### Problem2: symlinking conda

It seems there is a "symlink_conda" function, that replaces the newly generated 3 files by simlinks and doesn't include them. 

There was a hack that makes conda build by @mingwandroid here: #2167. But that logic got reverted in #2251 . Unfortunately adding back the hack doesnt work, because of problem 1 (the 3 files are not in always_include_files). So if problem 1 would be fixed, something would probably need to be done about problem 2, because uncoditionally calling symlink_conda will probably replace our files...

@msarahan Am I on the right track at all? Is the way of building conda still alright, or is there an alternative?

**Comment by neok-m4700** [comment]
```bash
$ conda info conda
conda 4.3.25 py36_0
-------------------
file name   : conda-4.3.25-py36_0.tar.bz2
name        : conda
version     : 4.3.25
build string: py36_0
build number: 0
channel     : defaults
size        : 510 KB
arch        : x86_64
date        : 2017-08-16
license     : BSD
md5         : 8d185c8a60c4355566b5ff01a5f2ba06
noarch      : None
platform    : linux
url         : https://repo.continuum.io/pkgs/free/linux-64/conda-4.3.25-py36_0.tar.bz2
dependencies:
    conda-env >=2.6
    pycosat >=0.6.1
    pyopenssl >=16.2
    python 3.6*
    requests >=2.12.4
    ruamel_yaml >=0.11.14

$ wget https://repo.continuum.io/pkgs/free/linux-64/conda-4.3.25-py36_0.tar.bz2
```
unpack the archive, the recipe is included in the archive, and in `<archive>/info/recipe/meta.yaml`, the header line has this: `# This file created by conda-build 3.0.11`

so I guess that building conda with conda-build 3.X.X is possible ?

**Comment by gabm** [comment]
yes seems possible indeed, the recipe is roughly like mine... 

for me its not working with the above problems...  Is there a build log on your build?

**Comment by msarahan** [comment]
It seems broken.  I don't know exactly how we've been building it - I haven't been the one doing that work, and you're the first to report this issue.  I'll look into it.

**Comment by msarahan** [comment]
I have hopefully addressed the conda issue with #2359
